src/main.py

```diff
@@ -580,24 +583,54 @@ def run(settings, components, variables):
   projectLength = getProjectLength(settings, components, v=v)
   vprint(v, 0, m, ' ... project length: {} years'.format(projectLength))
   projectCashflows = projectLifeCashflows(settings, components, lifetimeCashflows, projectLength, v=v)
+  compCashflows = {}
+  for comp in components:
+    tax = comp.getTax() if comp.getTax() is not None else settings.getTax()
+    inflation = comp.getInflation() if comp.getInflation() is not None else settings.getInflation()
+    compProjCashflows = projectComponentCashflows(comp, tax, inflation, lifetimeCashflows[comp.name], projectLength, v=v)
+    compCashflows[comp.name] = compProjCashflows
 
   vprint(v, 0, m, '='*90)
   vprint(v, 0, m, 'Economic Indicator Calculations')
   vprint(v, 0, m, '='*90)
   indicators = settings.getIndicators()
-  results = {}
-  if 'NPV_search' in indicators:
-    metric = npvSearch(settings, components, projectCashflows, projectLength, v=v)
-    results['NPV_mult'] = metric
-  if 'NPV' in indicators:
-    metric = NPV(components, projectCashflows, projectLength, settings.getDiscountRate(), v=v)
-    results['NPV'] = metric
-  if 'IRR' in indicators:
-    metric = IRR(components, projectCashflows, projectLength, v=v)
-    results['IRR'] = metric
-  if 'PI' in indicators:
-    metric = PI(components, projectCashflows, projectLength, settings.getDiscountRate(), v=v)
-    results['PI'] = metric
+  OutputType = settings.getOutput()
```

Comment:
I think `OutputType` should be `outputType` technically. I don't make these rules.